test/functional/tests/misc/test_flush_propagation.py

```diff
@@ -0,0 +1,112 @@
+        Logs._read_syslog(Logs.last_read_line)
+
+    with TestRun.step("Prepare devices for cache and core."):
+        cache_dev = TestRun.scsi_debug_devices[0]
+        core_dev = TestRun.disks["core"]
+        core_dev.create_partitions([Size(2, Unit.GibiByte)])
+        core_dev = core_dev.partitions[0]
+
+    with TestRun.step("Start cache on SCSI device and add core with xfs filesystem"):
+        cache = casadm.start_cache(cache_dev, CacheMode.WT)
+        core_dev.create_filesystem(Filesystem.xfs)
+        core = cache.add_core(core_dev)
+
+    with TestRun.step("Turn off cleaning policy and sequential cutoff"):
+        cache.set_cleaning_policy(CleaningPolicy.nop)
+        cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+
+    with TestRun.step("Mount exported object."):
+        if core.is_mounted():
+            core.unmount()
+        core.mount(mount_point)
+
+    with TestRun.step("Create temporary file on the exported object."):
+        tmp_file = create_random_test_file(f"{mount_point}/tmp.file", Size(1, Unit.GibiByte))
+        TestRun.executor.run("dmesg -C")
+        os_utils.sync()
+
+    with TestRun.step(f"Check {syslog_path} for flush request and delete temporary file."):
+        Logs.check_syslog_for_signals()
+        tmp_file.remove(True)
+
+
+@pytest.mark.os_dependent
+@pytest.mark.require_plugin("scsi_debug_fua_signals", dev_size_mb="8192", opts="1")
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+def test_flush_signal_propagation_cache():
+    """
+    title: Test for FLUSH signals propagation to core devuce
+    description: |
+      Test if OpenCAS propagates FLUSH signal to underlaying core device
+    pass_criteria:
+      - FLUSH requests should be propagated to core device.
+    """
+    with TestRun.step("Set mark in syslog to not read entries existing before the test."):
+        Logs._read_syslog(Logs.last_read_line)
+
+    with TestRun.step("Prepare devices for cache and core."):
+        cache_dev = TestRun.disks["cache"]
+        core_dev = TestRun.scsi_debug_devices[0]
+        cache_dev.create_partitions([Size(2, Unit.GibiByte)])
+        cache_dev = cache_dev.partitions[0]
+
+    with TestRun.step("Start cache and add SCSI device with xfs filesystem as core."):
+        cache = casadm.start_cache(cache_dev, CacheMode.WT)
+        core_dev.create_filesystem(Filesystem.xfs)
+        core = cache.add_core(core_dev)
+
+    with TestRun.step("Turn off cleaning policy and sequential cutoff"):
+        cache.set_cleaning_policy(CleaningPolicy.nop)
+        cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+
+    with TestRun.step("Mount exported object."):
+        if core.is_mounted():
+            core.unmount()
+        core.mount(mount_point)
+
+    with TestRun.step("Create temporary file on the exported object."):
+        tmp_file = create_random_test_file(f"{mount_point}/tmp.file", Size(1, Unit.GibiByte))
+        TestRun.executor.run("dmesg -C")
```

Comment:
what is the purpose of cleaning logs here?

Reply:
Scsi debug module gathers info about flush request from dmesg. I've added cleaning dmesg to ensure that the flush requests where issued during the actual flush of data to the disk, not i.e. during creating partitions or fs

Reply:
Can't you just do it with `Logs._read_syslog(Logs.last_read_line)` like a few lines above?

Reply:
Done